src/actions/marketo/marketo_transaction.ts

```diff
@@ -70,15 +71,23 @@ export class MarketoTransaction {
 
     this.marketo = this.marketoClientFromRequest(request)
 
-    const queue = new Queue()
-
     let rows: Hub.JsonDetail.Row[] = []
-
-    const sendChunk = () => {
-      const chunk = rows.slice(0)
-      const task = async () => this.processChunk(chunk)
-      rows = []
-      queue.addTask(task)
+    const results: Result[] = []
+    const errors: Result[] = []
+
+    const sendChunk = async () => {
+      return new Promise<void>((resolve, reject) => {
+        const chunk = rows.slice(0)
+        rows = []
+        this.processChunk(chunk).then((response: any) => {
+          results.push(response)
+          resolve()
```

Comment:
probably `resolve(response)`? You might not be using it here but that's what I'd expect from the interface.

Reply:
This is fine because it is a void Promise